packages/instantsearch-ui-components/src/lib/cx.ts

```diff
@@ -1,5 +1,13 @@
-export function cx(
-  ...classNames: Array<string | number | boolean | undefined | null>
-) {
-  return classNames.filter(Boolean).join(' ');
+type ClassValue = string | undefined | boolean | null | number;
+
+export function cx(...classNames: Array<ClassValue | ClassValue[]>) {
```

Comment:
too bad we need this, if I remember it correctly it's a significant performance bottleneck to make cx do more work. Where are we calling it with more than straight values?

Reply:
In InstantSearch.js we allow passing `cssClasses` to widgets as `string | string[]` so we need to flatten.

Note that this change actually makes `cx` behave exactly as it does in `@algolia/ui-components` so in InstantSearch.js, we already had this performande impact—this PR doesn't change it.

Reply:
👍 I see, I thought we did the extra flattening in the widget, but this makes sense then.